client/src/AuthContext.tsx

```diff
@@ -0,0 +1,98 @@
+import { createContext, useState, useEffect, useContext, ReactNode } from 'react';
+import axios from 'axios';
+
+interface Adopter {
+  id: string;
+  username: string;
+  role: string;
+}
+
+interface Shelter extends Adopter {
+  email: string;
```

Comment:
Email is the same as username i guess the additional interface for Shelter is unnecessary

Reply:
it is needed.